crates/iroha/tests/integration/events/data.rs

```diff
@@ -124,105 +127,84 @@ fn wasm_execution_should_produce_events() -> Result<()> {
         isi_calls = isi_calls
     );
 
-    transaction_execution_should_produce_events(
-        WasmSmartContract::from_compiled(wat.into_bytes()),
-        10_615,
-    )
+    transaction_execution_should_produce_events(WasmSmartContract::from_compiled(wat.into_bytes()))
+        .await
 }
 
-fn transaction_execution_should_produce_events(
-    executable: impl Into<Executable>,
-    port: u16,
+async fn transaction_execution_should_produce_events(
+    executable: impl Into<Executable> + Send,
 ) -> Result<()> {
-    let (_rt, _peer, client) = <PeerBuilder>::new().with_port(port).start_with_runtime();
-    wait_for_genesis_committed(&[client.clone()], 0);
-
-    // spawn event reporter
-    let listener = client.clone();
-    let (init_sender, init_receiver) = mpsc::channel();
-    let (event_sender, event_receiver) = mpsc::channel();
-    let event_filter = DataEventFilter::Any;
-    thread::spawn(move || -> Result<()> {
-        let event_iterator = listener.listen_for_events([event_filter])?;
-        init_sender.send(())?;
-        for event in event_iterator {
-            event_sender.send(event)?
-        }
-        Ok(())
-    });
-
-    // submit transaction to produce events
-    init_receiver.recv()?;
-    let transaction = client.build_transaction(executable, Metadata::default());
-    client.submit_transaction_blocking(&transaction)?;
-
-    // assertion
-    iroha_logger::info!("Listening for events");
-    for i in 0..4_usize {
-        let event: DataEvent = event_receiver.recv()??.try_into()?;
-        iroha_logger::info!("Event: {:?}", event);
-        assert!(matches!(event, DataEvent::Domain(_)));
-        if let DataEvent::Domain(domain_event) = event {
-            assert!(matches!(domain_event, DomainEvent::Created(_)));
-
-            if let DomainEvent::Created(created_domain) = domain_event {
-                let domain_id = DomainId::new(i.to_string().parse().expect("Valid"));
-                assert_eq!(domain_id, *created_domain.id());
-            }
-        }
+    let network = NetworkBuilder::new().start().await?;
+    let mut events_stream = network
+        .client()
+        .listen_for_events_async([DataEventFilter::Any])
+        .await?;
+
+    {
+        let client = network.client();
+        let tx = client.build_transaction(executable, <_>::default());
+        spawn_blocking(move || client.submit_transaction_blocking(&tx)).await??;
```

Comment:
This seems to be a common pattern. Is the goal to delegate blocking tasks to the dedicated thread pool to reduce overall test time?

It is fine to me if this is a workaround until #3130 async client. Otherwise, developers may not want to be aware of concurrency with other tests and may forget to wrap a blocking task in `spawn_blocking`

Reply:
I just read the review notes now

Reply:
> I just read the review notes now

Yeah (

It is the best temporary solution I found until the async client is there. Though, I could have introduced some way to reduce boilerplate of `clone`+`spawn_blocking`+`move`, like an extension trait to the `Client`, I thought it's better to keep tests very simple and rely on `tokio` utils directly as much as possible.